sdk/storage/Azure.Storage.Files.Shares/src/ShareErrors.cs

```diff
@@ -17,18 +17,18 @@ internal static class ShareErrors
             string shareClient) =>
             new InvalidOperationException($"{leaseClient} requires either a {fileClient} or {shareClient}");
 
-        public static void AssertAlgorithmSupport(ValidationAlgorithm? algorithm)
+        public static void AssertAlgorithmSupport(StorageChecksumAlgorithm? algorithm)
         {
-            ValidationAlgorithm resolved = (algorithm ?? ValidationAlgorithm.None).ResolveAuto();
+            StorageChecksumAlgorithm resolved = (algorithm ?? StorageChecksumAlgorithm.None).ResolveAuto();
             switch (resolved)
             {
-                case ValidationAlgorithm.None:
-                case ValidationAlgorithm.MD5:
+                case StorageChecksumAlgorithm.None:
+                case StorageChecksumAlgorithm.MD5:
                     return;
-                case ValidationAlgorithm.StorageCrc64:
+                case StorageChecksumAlgorithm.StorageCrc64:
                     throw new ArgumentException("Azure File Shares do not support CRC-64.");
                 default:
-                    throw new ArgumentException($"SDK does not support ValidationAlgorithm value {Enum.GetName(typeof(ValidationAlgorithm), resolved)}.");
+                    throw new ArgumentException($"SDK does not support ValidationAlgorithm value {Enum.GetName(typeof(StorageChecksumAlgorithm), resolved)}.");
```

Comment:
Nit - avoid using the term SDK in error messages.  I'm not sure the message is helpful as-is because if I pass a bogus value it won't be able to print anything.  Maybe we want something more like:
```suggestion
                    throw new ArgumentException($"{nameof(StorageChecksumAlgorithm)} does not support value {Enum.GetName(typeof(StorageChecksumAlgorithm), resolved) ?? resolved}.");
```